syntax = "proto3";

message FileInfo {
    string name = 1;
    int64 size = 2;
    string timestamp = 3;
}

message ListFilesResponse {
    repeated FileInfo file_info = 1;
}

message FindFileRequest {
    string file_name = 1;
}
  
message FindFileResponse {
    FileInfo file_info = 1;
}

service FileServices{
    rpc ListFiles(FileInfo) returns (ListFilesResponse) {}
    rpc FindFile(FindFileRequest) returns (FindFileResponse);
}